sklearn/preprocessing/discretization.py

```diff
@@ -0,0 +1,261 @@
+        -np.inf,
+        offset_[i] + bin_width_[i] * np.arange(1, n_bins_[i]),
+        np.inf
+      ])
+
+    See also
+    --------
+     sklearn.preprocessing.Binarizer : class used to bin values as ``0`` or
+        ``1`` based on a parameter ``threshold``.
+    """
+
+    def __init__(self, n_bins=2, ignored_features=None):
+        self.n_bins = n_bins
+        self.ignored_features = ignored_features
+
+    def fit(self, X, y=None):
+        """Fits the estimator.
+
+        Parameters
+        ----------
+        X : numeric array-like, shape (n_samples, n_features)
+            Data to be discretized.
+
+        y : ignored
+
+        Returns
+        -------
+        self
+        """
+        X = check_array(X, dtype=FLOAT_DTYPES)
+
+        n_features = X.shape[1]
+        ignored = self._check_ignored_features(self.ignored_features,
+                                               n_features)
+        self.transformed_features_ = np.delete(np.arange(n_features), ignored)
+
+        offset = np.min(X, axis=0)
+        offset[ignored] = 0
+        self.offset_ = offset
+
+        n_bins = self._check_n_bins(self.n_bins, n_features, ignored)
+        n_bins[ignored] = 0
+        self.n_bins_ = n_bins
+
+        ptp = np.ptp(X, axis=0)
+        same_min_max = np.where(ptp == 0)[0]
+        if len(same_min_max) > 0:
+            warnings.warn("Features {} are constant and will be replaced with "
+                          "0.".format(", ".join(str(i) for i in same_min_max)))
+
+        with np.errstate(divide='ignore', invalid='ignore'):
+            bin_widths = ptp / self.n_bins_
+        bin_widths[ignored] = 0
+        self.bin_width_ = bin_widths
+        return self
+
+    @staticmethod
+    def _check_n_bins(n_bins, n_features, ignored):
+        if isinstance(n_bins, numbers.Number):
+            if n_bins < 2:
+                raise ValueError("Discretizer received an invalid number "
+                                 "of bins. Received {0}, expected at least 2."
+                                 .format(n_bins))
+            return np.ones(n_features) * n_bins
+
+        n_bins = check_array(n_bins, dtype=int, copy=True, ensure_2d=False)
+
+        if n_bins.ndim > 1 or n_bins.shape[0] != n_features:
+            raise ValueError("n_bins must be a scalar or array "
+                             "of shape (n_features,).")
+
+        bad_nbins_value = n_bins < 2
+        bad_nbins_value[ignored] = False
+
+        violating_indices = np.where(bad_nbins_value)[0]
+        if violating_indices.shape[0] > 0:
+            indices = ", ".join(str(i) for i in violating_indices)
+            raise ValueError("Discretizer received an invalid number "
+                             "of bins at indices {}. Number of bins "
+                             "must be at least 2."
+                             .format(indices))
+        return n_bins
+
+    @staticmethod
+    def _check_ignored_features(ignored, n_features):
```

Comment:
I encounter more often `_validate_(self)` method in the code base and I would strive for consistency with the package than personal style choice ;)